tx.go

```diff
@@ -587,6 +589,31 @@ func (tx *Tx) Page(id int) (*common.PageInfo, error) {
 	return info, nil
 }
 
+// seekWithSharedCursorThreadUnsafe should be used in internal functions (for thread safe version look at seekWithSharedCursorThreadSafe).
+// Reuses []elemRef stored in Tx to remove unnecessary allocations.
+func (tx *Tx) seekWithSharedCursorThreadUnsafe(b *Bucket, seek []byte) (key []byte, value []byte, flags uint32, cursor Cursor) {
+	// should be 100% compatible with Bucket.Cursor()
+	cursor = Cursor{bucket: b, stack: tx.reusableRef}
+	key, value, flags = cursor.seek(seek)
+	tx.reusableRef = cursor.stack[:0]
+	return
+}
+
+// seekWithSharedCursorThreadSafe should be used in internal functions.
+// Reuses []elemRef stored in Tx to remove unnecessary allocations.
+// It doesn't return a cursor because it's not thread safe
+func (tx *Tx) seekWithSharedCursorThreadSafe(b *Bucket, seek []byte) (key []byte, value []byte, flags uint32) {
+	if atomic.CompareAndSwapUint32(&tx.reusableRefLocked, 0, 1) {
+		// should be 100% compatible with Bucket.Cursor()
+		c := Cursor{bucket: b, stack: tx.reusableRef}
+		key, value, flags = c.seek(seek)
+		tx.reusableRef = c.stack[:0]
```

Comment:
4 lines above are duplicated. Can you call `seekWithSharedCursorThreadUnsafe` from here?